modules/appeals_api/app/services/appeals_api/file_validator.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'pdf_info'
+
+module AppealsApi
+  class FileValidator
+    include SentryLogging
+
+    class UploadValidationError < StandardError; end
+
+    MAX_PAGE_SIZE = { width: 11, height: 11 }.freeze
+    MAX_FILE_SIZE = 100.megabytes
+    MAX_SIZE_STRING = '100 MB'
+
+    def initialize(file)
+      @file = file
+      @filename = File.basename(@file)
+    end
+
+    def call
+      return file_type_error unless pdf_metadata_present?
+      return file_size_error unless valid_file_size?
+      return max_dimension_error unless valid_page_dimensions?
+
+      validated
```

Comment:
Why extract out a lone `true` call with a method? Just return `true` if that's what you want to do.

What do you think about returning a two-element array of status & data instead of this split brain problem where the caller needs to worry about data types since errors return a hash but success returns a boolean `true`?

```suggestion
return [:error, file_type_error] unless pdf_metadata_present?
return [:error, file_size_error] unless valid_file_size?
return [:error, max_dimension_error] unless valid_page_dimensions?

[:ok, {}]
```

Then the caller can do a `status, data = FileValidator.new(myfile).call` and know via `status` whether the result was ok or an error, and then work with the `data` if it needs to.

Reply:
This makes me wish Ruby had better support for tuples. I agree with the premise, however - The interface should probably be consistent so that there's not a cycle of `if` statements to parse out the response.

Reply:
The lone true call: me playing around with the return of a successful validation. I'm having trouble putting the bigger picture together here. What is he ultimate responsibility of this validator, and how we want to handle its output in the controller?

Reply:
Said simply, I think the responsibility of the validator is to accept a document, and determine whether it passes the criteria or not.

As for handling the output of the controller, we shouldn't be assuming that the Validator's lone client will be a controller. In theory, we could call this interface anywhere.